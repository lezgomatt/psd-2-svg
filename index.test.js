const { convertFile } = require('.');

test('converts asterisk', () => {
  let svgResult = convertFile('./testdata/asterisk.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts bed', () => {
  let svgResult = convertFile('./testdata/bed.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts like', () => {
  let svgResult = convertFile('./testdata/like.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts shapes', () => {
  let svgResult = convertFile('./testdata/shapes.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts layers', () => {
  let svgResult = convertFile('./testdata/layers.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts strokes', () => {
  let svgResult = convertFile('./testdata/strokes.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts lines', () => {
  let svgResult = convertFile('./testdata/lines.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts stroke-pos', () => {
  let svgResult = convertFile('./testdata/stroke-pos.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts dashes', () => {
  let svgResult = convertFile('./testdata/dashes.psd').toString();
  expect(svgResult).toMatchSnapshot();
});

test('converts special-name', () => {
  let svgResult = convertFile('./testdata/special-name.psd').toString();
  expect(svgResult).toMatchSnapshot();
});
