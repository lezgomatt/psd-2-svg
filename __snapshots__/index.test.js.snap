// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts asterisk 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"640\\" height=\\"480\\">
	<path class=\\"Layer_1\\" fill=\\"rgb(0, 0, 0)\\" fill-rule=\\"evenodd\\"
		d=\\"M388.4092 240 C388.4092 240,435.7968 214.3333,435.7968 214.3333 C441.2582 211.4453,444.7933 207.1406,446.3967 201.4166 C448 195.6953,447.2568 190.2786,444.1698 185.1667 C444.1698 185.1667,432.7682 166.8333,432.7682 166.8333 C429.6785 161.7239,425.0772 158.4166,418.9617 156.9167 C412.8434 155.4167,407.0536 156.112,401.5922 159 C401.5922 159,354.2046 184.5,354.2046 184.5 C354.2046 184.5,354.2046 133.3333,354.2046 133.3333 C354.2046 127.5573,351.9471 122.5573,347.4349 118.3333 C342.92 114.112,337.5755 112,331.4015 112 C331.4015 112,308.5984 112,308.5984 112 C302.4217 112,297.0772 114.112,292.565 118.3333 C288.0501 122.5573,285.7954 127.5573,285.7954 133.3333 C285.7954 133.3333,285.7954 184.5,285.7954 184.5 C285.7954 184.5,238.4078 159,238.4078 159 C232.9436 156.112,227.1538 155.4167,221.0382 156.9167 C214.92 158.4166,210.3187 161.7239,207.2317 166.8333 C207.2317 166.8333,195.8302 185.1667,195.8302 185.1667 C192.7404 190.2786,192 195.6953,193.6033 201.4166 C195.2067 207.1406,198.739 211.4453,204.2032 214.3333 C204.2032 214.3333,251.5908 240,251.5908 240 C251.5908 240,204.2032 265.6667,204.2032 265.6667 C198.739 268.5573,195.2067 272.862,193.6033 278.5833 C192 284.3073,192.7404 289.7239,195.8302 294.8333 C195.8302 294.8333,207.2317 313.1667,207.2317 313.1667 C210.3187 318.2786,214.92 321.5833,221.0382 323.0833 C227.1538 324.5833,232.9436 323.8906,238.4078 321 C238.4078 321,285.7954 295.5,285.7954 295.5 C285.7954 295.5,285.7954 346.6667,285.7954 346.6667 C285.7954 352.4427,288.0501 357.4427,292.565 361.6666 C297.0772 365.888,302.4217 368,308.5984 368 C308.5984 368,331.4015 368,331.4015 368 C337.5755 368,342.92 365.888,347.4349 361.6666 C351.9471 357.4427,354.2046 352.4427,354.2046 346.6667 C354.2046 346.6667,354.2046 295.5,354.2046 295.5 C354.2046 295.5,401.5922 321,401.5922 321 C407.0536 323.8906,412.8434 324.5833,418.9617 323.0833 C425.0772 321.5833,429.6785 318.2786,432.7682 313.1667 C432.7682 313.1667,444.1698 294.8333,444.1698 294.8333 C447.2568 289.7239,448 284.3073,446.3967 278.5833 C444.7933 272.862,441.2582 268.5573,435.7968 265.6667 C435.7968 265.6667,388.4092 240,388.4092 240 Z\\"/>
</svg>"
`;

exports[`converts bed 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"640\\" height=\\"480\\">
	<path class=\\"Shape_3\\" fill=\\"rgb(0, 0, 0)\\" fill-rule=\\"evenodd\\"
		d=\\"M263.75 195 C263.75 193.6462,263.2853 192.4743,262.3584 191.4843 C261.4302 190.4956,260.3316 190,259.0625 190 C259.0625 190,249.6875 190,249.6875 190 C248.4172 190,247.3186 190.4956,246.3916 191.4843 C245.4634 192.4743,245 193.6462,245 195 C245 195,245 290,245 290 C245 290,263.75 290,263.75 290 C263.75 290,263.75 270,263.75 270 C263.75 270,376.25 270,376.25 270 C376.25 270,376.25 290,376.25 290 C376.25 290,395 290,395 290 C395 290,395 255,395 255 C395 253.6462,394.5353 252.4743,393.6084 251.4843 C392.6803 250.4956,391.5816 250,390.3125 250 C390.3125 250,263.75 250,263.75 250 C263.75 250,263.75 195,263.75 195 Z
		M287.1875 245 C282.0113 245,277.5927 243.0469,273.9306 239.1406 C270.2686 235.2344,268.4375 230.5212,268.4375 225 C268.4375 219.48,270.2686 214.7656,273.9306 210.8594 C277.5927 206.9531,282.0113 205,287.1875 205 C292.3625 205,296.7822 206.9531,300.4443 210.8594 C304.1064 214.7656,305.9375 219.48,305.9375 225 C305.9375 230.5212,304.1064 235.2344,300.4443 239.1406 C296.7822 243.0469,292.3625 245,287.1875 245 Z
		M310.625 245 C310.625 245,310.625 215,310.625 215 C310.625 213.6462,311.0885 212.4744,312.0166 211.4844 C312.9436 210.4956,314.0422 210,315.3125 210 C315.3125 210,366.875 210,366.875 210 C374.6386 210,381.2671 212.9297,386.7603 218.789 C392.2534 224.6484,395 231.7187,395 240 C395 240,395 245,395 245 C395 245,310.625 245,310.625 245 Z\\"/>
</svg>"
`;

exports[`converts layers 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"640\\" height=\\"480\\">
	<path class=\\"Hidden\\" visibility=\\"hidden\\" fill=\\"rgb(115, 46, 240)\\" fill-rule=\\"evenodd\\"
		d=\\"M321 241 C321 241,421 241,421 241 C421 241,421 341,421 341 C421 341,321 341,321 341 C321 341,321 241,321 241 Z\\"/>
	<g class=\\"Folder\\">
		<path class=\\"Bottom\\" fill=\\"rgb(49, 238, 134)\\" fill-rule=\\"evenodd\\"
			d=\\"M296 215 C296 215,396 215,396 215 C396 215,396 315,396 315 C396 315,296 315,296 315 C296 315,296 215,296 215 Z\\"/>
		<g class=\\"Nested\\">
			<path class=\\"Inner\\" fill=\\"rgb(181, 245, 29)\\" fill-rule=\\"evenodd\\"
				d=\\"M271 189 C271 189,371 189,371 189 C371 189,371 289,371 289 C371 289,271 289,271 289 C271 289,271 189,271 189 Z\\"/>
		</g>
		<path class=\\"Middle\\" fill=\\"rgb(247, 171, 40)\\" fill-rule=\\"evenodd\\"
			d=\\"M245 161 C245 161,345 161,345 161 C345 161,345 261,345 261 C345 261,245 261,245 261 C245 261,245 161,245 161 Z\\"/>
	</g>
	<path class=\\"Top\\" fill=\\"rgb(38, 116, 246)\\" fill-rule=\\"evenodd\\"
		d=\\"M220 140 C220 140,320 140,320 140 C320 140,320 240,320 240 C320 240,220 240,220 240 C220 240,220 140,220 140 Z\\"/>
</svg>"
`;

exports[`converts like 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"640\\" height=\\"480\\">
	<path class=\\"Shape_3\\" fill=\\"rgb(0, 0, 0)\\" fill-rule=\\"evenodd\\"
		d=\\"M239.75 319.565 C237.3336 319.565,235.2438 318.695,233.4805 316.9529 C231.715 315.213,230.8333 313.1509,230.8333 310.7666 C230.8333 308.3844,231.715 306.3223,233.4805 304.5802 C235.2438 302.8403,237.3336 301.9682,239.75 301.9682 C242.1642 301.9682,244.254 302.8403,246.0195 304.5802 C247.7828 306.3223,248.6666 308.3844,248.6666 310.7666 C248.6666 313.1509,247.7828 315.213,246.0195 316.9529 C244.254 318.695,242.1642 319.565,239.75 319.565 Z
		M406.1712 242.7164 C404.1728 246.933,401.6889 249.0875,398.7174 249.1777 C400.1107 250.7372,401.271 252.9132,402.2005 255.7078 C403.1279 258.5045,403.5937 261.0478,403.5937 263.3377 C403.5937 269.6615,401.1316 275.1154,396.2096 279.6972 C397.8815 282.6314,398.7174 285.7934,398.7174 289.183 C398.7174 292.5747,397.9033 295.9429,396.2793 299.2874 C394.6531 302.6341,392.4479 305.0399,389.6614 306.5048 C390.1251 309.2544,390.358 311.8213,390.358 314.2034 C390.358 329.5104,381.4414 337.1618,363.6081 337.1618 C363.6081 337.1618,346.75 337.1618,346.75 337.1618 C334.581 337.1618,318.6982 333.8173,299.1015 327.1261 C298.6357 326.9435,297.2903 326.4623,295.0612 325.6826 C292.832 324.905,291.1819 324.3315,290.1152 323.9642 C289.0464 323.599,287.4202 323.0706,285.2389 322.3832 C283.0555 321.6958,281.29 321.1932,279.9446 320.871 C278.5971 320.5509,277.0646 320.2524,275.347 319.9774 C273.6272 319.7025,272.1643 319.565,270.9583 319.565 C270.9583 319.565,266.5 319.565,266.5 319.565 C266.5 319.565,266.5 231.5809,266.5 231.5809 C266.5 231.5809,270.9583 231.5809,270.9583 231.5809 C272.443 231.5809,274.0931 231.1685,275.9043 230.3436 C277.7155 229.5188,279.5724 228.2815,281.4772 226.6318 C283.3798 224.9821,285.1693 223.356,286.8411 221.7514 C288.513 220.149,290.3699 218.132,292.4141 215.7025 C294.456 213.2752,296.0582 211.3269,297.2207 209.8598 C298.381 208.3949,299.8439 206.5153,301.6093 204.2233 C303.3727 201.9335,304.4415 200.5588,304.8138 200.0991 C309.9208 193.8676,313.4975 189.6982,315.5416 187.5888 C319.3491 183.6493,322.1116 178.6315,323.8314 172.5353 C325.549 166.4413,326.9661 160.691,328.0807 155.2822 C329.1953 149.8756,330.9586 145.9811,333.375 143.5968 C342.2916 143.5968,348.2346 145.7513,351.2083 150.0581 C354.1798 154.3671,355.6667 161.011,355.6667 169.992 C355.6667 175.4008,353.4375 182.7557,348.9791 192.0568 C344.5208 201.36,342.2916 208.6698,342.2916 213.9841 C342.2916 213.9841,391.3333 213.9841,391.3333 213.9841 C395.9767 213.9841,400.1107 215.7498,403.7331 219.2769 C407.3555 222.8061,409.1666 226.9067,409.1666 231.5809 C409.1666 234.7901,408.1675 238.5019,406.1712 242.7164 Z
		M416.4114 206.8354 C409.3517 199.8714,400.9923 196.3873,391.3333 196.3873 C391.3333 196.3873,366.8125 196.3873,366.8125 196.3873 C371.2708 187.3139,373.5 178.5155,373.5 169.992 C373.5 159.269,371.8738 150.7455,368.6237 144.4217 C365.3714 138.0978,360.6344 133.4473,354.4127 130.4679 C348.1889 127.4907,341.1771 126,333.375 126 C328.638 126,324.4583 127.697,320.8359 131.0865 C316.8413 134.9359,313.9612 139.885,312.1979 145.9339 C310.4324 151.9828,309.0152 157.7804,307.9485 163.3245 C306.8797 168.8708,305.2317 172.7888,303.0026 175.0786 C298.4507 179.9375,293.4807 185.8017,288.0951 192.6754 C278.7125 204.683,272.3516 211.7845,269.0078 213.9841 C269.0078 213.9841,230.8333 213.9841,230.8333 213.9841 C225.9091 213.9841,221.7077 215.7025,218.2246 219.1394 C214.7415 222.5763,213 226.7242,213 231.5809 C213 231.5809,213 319.565,213 319.565 C213 324.4239,214.7415 328.5696,218.2246 332.0065 C221.7077 335.4434,225.9091 337.1618,230.8333 337.1618 C230.8333 337.1618,270.9583 337.1618,270.9583 337.1618 C273.0003 337.1618,279.4091 338.9941,290.1849 342.6608 C301.6093 346.6003,311.6624 349.6033,320.3483 351.6654 C329.032 353.7276,337.8333 354.7586,346.75 354.7586 C346.75 354.7586,359.707 354.7586,359.707 354.7586 C359.707 354.7586,364.7226 354.7586,364.7226 354.7586 C377.819 354.7586,388.3596 351.0683,396.349 343.6919 C404.3361 336.3155,408.2828 326.2562,408.1914 313.5161 C413.7643 306.4597,416.5508 298.3036,416.5508 289.0455 C416.5508 287.0306,416.4114 285.0587,416.1328 283.1341 C419.6616 276.9949,421.4271 270.3961,421.4271 263.3377 C421.4271 260.0383,421.0091 256.8763,420.1731 253.8519 C424.7229 247.0705,427 239.6017,427 231.4434 C427 222.0049,423.469 213.8015,416.4114 206.8354 Z\\"/>
</svg>"
`;

exports[`converts shapes 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"640\\" height=\\"480\\">
	<path class=\\"Shape_1\\" opacity=\\"0.8\\" fill=\\"rgb(241, 29, 99)\\" fill-rule=\\"evenodd\\"
		d=\\"M166 140 C166 140,366 140,366 140 C366 140,366 340,366 340 C366 340,166 340,166 340 C166 340,166 140,166 140 Z\\"/>
	<path class=\\"Shape_2\\" opacity=\\"0.65\\" fill=\\"rgb(156, 244, 31)\\" fill-rule=\\"evenodd\\"
		d=\\"M366 340 C310.7 340,266 295.3,266 240 C266 184.7,310.7 140,366 140 C421.3 140,466 184.7,466 240 C466 295.3,421.3 340,366 340 Z\\"/>
	<path class=\\"Layer_1\\" opacity=\\"0.5\\" fill=\\"rgb(61, 108, 242)\\" fill-rule=\\"evenodd\\"
		d=\\"M365 147 C365 147,473.2532 334.5,473.2532 334.5 C473.2532 334.5,256.7468 334.5,256.7468 334.5 C256.7468 334.5,365 147,365 147 Z\\"/>
</svg>"
`;

exports[`converts strokes 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"640\\" height=\\"480\\">
	<path class=\\"Shape_1\\" fill=\\"transparent\\" stroke=\\"rgb(0, 255, 0)\\" stroke-width=\\"1\\" stroke-linejoin=\\"round\\" fill-rule=\\"evenodd\\"
		d=\\"M259 179 C259 179,409 179,409 179 C409 179,409 329,409 329 C409 329,259 329,259 329 C259 329,259 179,259 179 Z\\"/>
	<path class=\\"Shape_1_copy\\" fill=\\"transparent\\" stroke=\\"rgb(0, 0, 255)\\" stroke-width=\\"1\\" fill-rule=\\"evenodd\\"
		d=\\"M232 152 C232 152,382 152,382 152 C382 152,382 302,382 302 C382 302,232 302,232 302 C232 302,232 152,232 152 Z\\"/>
</svg>"
`;
